build-system/tasks/performance/measure-documents.js

```diff
@@ -148,32 +243,43 @@ function writeMetrics(url, version, metrics) {
  * @param {string} url
  * @param {string} version "control" or "experiment"
  * @param {Object} options
+ * @param {Object} handlers
  * @return {Promise}
  */
-async function measureDocument(url, version, {headless}) {
+async function measureDocument(url, version, {headless}, handlers) {
   const browser = await puppeteer.launch({
     headless,
     args: [
       '--allow-file-access-from-files',
       '--enable-blink-features=LayoutInstabilityAPI',
+      '--disable-web-security',
     ],
   });
 
   const page = await browser.newPage();
+  const {handlerOptions, handlerName} = getHandlerFromUrl(url, handlers);
   await page.setCacheEnabled(false);
   await setupMeasurement(page);
 
+  if (handlerOptions) {
+    await page.setRequestInterception(true);
+    setUpRequestHandler(handlerOptions, handlerName, page);
+  }
+
   try {
     await page.goto(`file:${urlToCachePath(url, version)}`, {
       waitUntil: 'networkidle0',
     });
-  } catch {
+  } catch (e) {
     // site did not load
+    console.log(e);
```

Comment:
@kevinkimball I found this really helpful when debugging and it could be helpful for other teams as they create the own request handlers. Do you think it's worth keeping?

Reply:
Seems OK to me, or could be hidden behind a logging flag

Reply:
If it didn't load, the test will fail right? How about throwing an error after closing the browser?

Reply:
I think a log flag would be good, and then logging the error after the browser closes.